### PR TITLE
General refactor

### PR DESCRIPTION
**Pull request by KevinKra**
-Renamed methods to be more descriptive
-Renamed App state to be more descriptive
-Utilize rest parameter on passed props
-Comment out properties passed but not used
-Leave descriptive comments regarding method renames

+Structs || data flow not modified

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -40,105 +41,119 @@ class App extends React.Component {
       .catch(err => console.log(err));
   }
 
-  updateResults = query => {
+  //does App need to know if the suggestions bar is open or closed?
```
Comment (jacobogart) [inline src/components/App.js]: Good catch! As far as I know, the only thing that should make the suggestions appear is typing in the search bar, and the only thing that should make them go away is clicking a suggestion, which lives in the search bar. There could be edge cases where a user starts typing but then decides to click something else, but that could probably be solved by a "focus out" event listener on the search bar component, which would cover suggestions too.

TLDR, I am pro moving this piece of state to Search Bar

File: src/components/App.js [hunk 2]
```diff
@@ -40,105 +41,119 @@ class App extends React.Component {
...
-    let searchResultsComponent = 
-      <SearchResults 
-        toggleSplash={this.toggleSplash}
+    let searchResultsComponent = (
+      <SearchResults
+        displaySplashPage={this.displaySplashPage}
         searchResults={this.state.searchResults}
-        toggleResults={this.toggleResults}
-        selectResult={this.selectResult}
+        // there is no displayAllSearchResults || toggleResults used in this component
```
Comment (jacobogart) [inline src/components/App.js]: It appears so, but we can check with @tiffanybacher tomorrow since this is mostly her component.

File: src/components/App.js [hunk 3]
```diff
@@ -40,105 +41,119 @@ class App extends React.Component {
...
+            updateSearchResults={this.updateSearchResults}
+            selectSearchResult={this.selectSearchResult}
             searchResults={this.state.searchResults}
-            showSuggestions={this.state.showSuggestions}
+            displaySearchSuggestions={this.state.displaySearchSuggestions}
           />
         </header>
         <section className="App">
-          <article className="mainContent">
-            {card}
```
Comment (jacobogart) [inline src/components/App.js:140]: I think this should be kept as three lines. It's a little hard to find {card} when it's nested between the article tags in one line.

File: src/components/Gallery.js [hunk 4]
```diff
@@ -1,22 +1,27 @@
...
+import "../css/Gallery.css";
 
 export default class Gallery extends Component {
   constructor(props) {
     super(props);
-    this.state = {
-
-    };
+    this.state = {};
+    this.shows = this.props.cardData.shows || [];
```
Comment (jacobogart) [inline src/components/Gallery.js:11]: I think this piece of logic should actually be a local variable within the render function. So far in our limit interaction with React I haven't seen any property of the class other than state. I'm not sure what sort of effect this will have on automatic rerenders. When state is changed I know it will call the render function again, but I don't know if it will call the constructor as well, which is where this property is being assigned. Let me know if you found anything out about this.

Comment (jacobogart) [inline src/components/Gallery.js:11]: We will also need a conditional here for determining card/data type. If it is a Bar or Queen it would be cardData.shows, but if it is a show, the data will live in either cardData.hosts (an array on queen or queens) or cardData.lineups (an object with properties being dates and the values being an array of queens), or both. I know the show one is a little tricky and I can pair with you if needed to work out that bit of logic.

File: src/components/Gallery.js [hunk 5]
```diff
@@ -1,22 +1,27 @@
...
   }
 
   render() {
+    // console.log(this.props.cardData);
     return (
       <section className="Gallery">
         <p>Gallery</p>
+        <ul>
+          {this.shows.map(show => {
+            return <GalleryImage showCategory={show.category} key={show.id} />;
```
Comment (jacobogart) [inline src/components/Gallery.js:20]: We would need to pass the imageURL as props for whatever we are iterating over so the image can be rendered in the thumbnail. We might also be able to reuse the Thumbnail component and just alter the styling so it can be used in the Gallery.

File: src/components/SearchBar.js [hunk 6]
```diff
@@ -11,39 +11,43 @@ class SearchBar extends Component {
...
       <section className="searchBarContainer">
         <form onSubmit={this.findResults} className="SearchBar">
+          {/* We cant click the submit button, we can only click the search results */}
           <input
             onKeyUp={this.handleChange}
             type="search"
             placeholder="Search..."
             className="searchTerm"
           />
+          {/* button isnt used currently */}
```
Comment (jacobogart) [inline src/components/SearchBar.js]: I'm confused by this comment. I am able to generate the search results page by either clicking the search button or pressing enter. Since the onSubmit event listener is in the form tag that both the input and button are wrapped it, it is listening for both.

File: src/components/NavBar.js [hunk 7]
```diff
@@ -9,17 +9,11 @@ class NavBar extends React.Component {
...
         <nav className="NavBar">
           <Hamburger />
-          <SearchBar
-            toggleResults={this.props.toggleResults}
-            updateResults={this.props.updateResults}
-            selectResult={this.props.selectResult}
-            searchResults={this.props.searchResults}
-            showSuggestions={this.props.showSuggestions}
-          />
+          <SearchBar {...this.props} />
```
Comment (jacobogart) [inline src/components/NavBar.js:16]: Good use of the spread operator when a component is just passing all of its props to a child. As we build out the hamburger menu it might cause some redundancies or props being passed and not used, but I don't think that's a major issue.

**Review by jacobogart — COMMENTED** [review]
Thank you for going through the code and helping us keep these semantic and streamlined! I added a few comments here and there why I was confused by a comment or change, but overall this should help keep the codebase more readable.

File: src/components/NavBar.js (hunk 7 above)
Comment (KevinKra) [reply src/components/NavBar.js:16]: I was thinking about that as well, I don't think it would be expensive to use the spread operator because props is an object and we are (I assume) passing references to certain parts of that object across multiple components. It's an interesting thought, I wonder if its accurate.

File: src/components/SearchBar.js (hunk 6 above)
Comment (KevinKra) [reply src/components/SearchBar.js]: That comment was a holdover that i forgot to remove, the submit/enter key do create a search page. When I first wrote it I thought the only way to get to pages was by clicking the results.

File: src/components/Gallery.js (hunk 5 above)
Comment (KevinKra) [reply src/components/Gallery.js:20]: of course! I was essentially returning the show.category to make sure that the correct data was actually rendering to the DOM. Once there are images they can just be swapped in, perhaps those could even be derived by the thumbnails.

File: src/components/Gallery.js (hunk 4 above)
Comment (KevinKra) [reply src/components/Gallery.js:11]: Good point, that piece may be a holdover when I was trying to tests things out so if it moves out out or gets removed entirely is fine. I am curious about that constructor question though, i wouldn't assume it would call the constructor either.

File: src/components/App.js (hunk 3 above)
Comment (KevinKra) [reply src/components/App.js:140]: Not sure why it looks like that in this log, it is still positioned normally in my local files.

File: src/components/App.js (hunk 2 above)
Comment (KevinKra) [reply src/components/App.js]: sure, we can just uncomment it if needed

File: src/components/App.js (hunk 1 above)
Comment (KevinKra) [reply src/components/App.js]: good solutions, we can move it around and reconfigure it hopefully pretty easily
